metals/src/main/scala/scala/meta/internal/metals/debug/DebugProvider.scala

```diff
@@ -84,13 +85,30 @@ class DebugProvider(
     statusBar: StatusBar,
     sourceMapper: SourceMapper,
     userConfig: () => UserConfiguration,
-) extends Cancelable {
+) extends Cancelable
+    with LogForwarder {
 
   import DebugProvider._
 
   private val debugSessions = new MutableCancelable()
 
-  override def cancel(): Unit = debugSessions.cancel()
+  private val currentRunner =
+    new ju.concurrent.atomic.AtomicReference[DebugRunner](null)
+
+  override def info(message: String): Unit = {
+    val runner = currentRunner.get()
+    if (runner != null) runner.stdout(message)
+  }
+
+  override def error(message: String): Unit = {
```

Comment:
Would it make sense to handle other message types from the `LogForwarder` as well, like `WARN`? :slightly_smiling_face:

Reply:
Warn will most likely come from the build server, we either get standard output or standard error here.